app/controllers/course/registrations_controller.rb

```diff
@@ -0,0 +1,111 @@
+class Course::RegistrationsController < Course::ModuleController
+  load_and_authorize_resource class: CourseUser.name
+  before_action :ensure_not_registered, only: :create
+  helper_method :localized_role
+
+  def index #:nodoc:
+    authorize! :manage, @course
```

Comment:
I intended to make the index the page where the course manager manages all registrations. The other actions have their permissions settled by cancan because there is a course user ID. I'm not sure how to make cancan prevent unauthorized access to the index page.